docs/development-guide/extending-gaffer.md

```diff
@@ -0,0 +1,217 @@
+for more details).
+
+#### Implementing an Interface
+
+The example we'll use here is the
+[`ToSet`](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/operation/impl/output/ToSet.html)
+Operation, this will takes an `Iterable` of `Elements` and converts it into a
+`Set`, a handy way to remove duplicate `Elements` if you're doing multiple hops
+and end up getting duplicates back.
+
+Taking a look at the [Javadoc](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/operation/impl/output/ToSet.html)
+for this class you should be able to see this class it implements both
+`InputOutput` (with an input type of `Iterable<? extends T>` and an output type
+of `Set<? extends T>` and `MultiInput<T>`). Within the class it also implements
+all the requisite methods eg, `shallowClone()`, `setInput()`, `getOptions()`
+etc.
+
+#### How to Write the Handler
+
+Using the same `ToSet` Operation from before, now that we have this definition
+of an operation, we need to actually tell Gaffer how to do it, we do this by
+defining either a single generic `Handler` and linking it to the Operation or in
+many cases we might want different implementations based on the store being used
+so we'd write a number of operation handlers and decide which one is needed at
+runtime.
+
+Here is the [`ToSetHandler`](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/store/operation/handler/output/ToSetHandler.html)
+which implements the appropriate `Handler` interface (in this case the
+`OutputOperationHandler`). It takes in the `ToSet` Operation that has been built
+with all the options etc. defined and then does the actual `doOperation()` logic to
+stream our Iterable's contents into a `LinkedHashSet`. This and most other
+handlers live inside the [Store library](./project-structure/components/store.md).
+
+#### How does Gaffer know which Handler to use?
+
+Handlers are registered in the `Store` class via the `addOpHandlers()` method,
+specifically for our example  `addOperationHandler(ToSet.class, new
+ToSetHandler<>())` is called to register it. Once this is done we can then
+use our `Operation` and its `Handler` in a user defined `OperationChain` to effect
+some manipulation/query on our graph data.
+
+You can alternatively use the an operations declarations config file (typically called
+`operationsDeclarations.json`) which can be used to define the Operation and its Handler
+like below:
+
+```json
+{
+    "operations": [
+        {
+            "operation": "uk.gov.gchq.gaffer.operation.impl.output.ToSet",
+            "handler": {
+                "class": "uk.gov.gchq.gaffer.store.operation.handler.output.ToSetHandler"
+            }
+        }
+    ]
+}
+```
+
+### Writing an Element/Object Generator
+
+There are two types of generators that define custom binding for taking a domain
+object and transforming it into a Gaffer `Element` for the `ElementGenerator` and
+likewise the reverse with an `ObjectGenerator`.
```

Comment:
This sentence is a bit odd. May be clearer to just state there are two types then explain what they are rather than try fit it in one confusing sentence.